contracts/cw20-merkle-airdrop/src/helpers.rs

```diff
@@ -42,3 +40,20 @@ impl CosmosSignature {
         Ok(addr)
     }
 }
+// Signature verification is done on external airdrop claims.
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+pub struct SignatureInfo {
+    pub claim_msg: Binary,
+    pub signature: Binary,
+}
+impl SignatureInfo {
+    pub fn extract_addr_from_memo(&self) -> Result<String, ContractError> {
+        let binary = Binary::from_base64(&self.claim_msg.to_string())?;
+        let memo_object = from_slice::<Memo>(&binary).unwrap();
+        Ok(memo_object.memo)
+    }
+}
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+pub struct Memo {
```

Comment:
We pass the information in the memo field but using memo name in the code is do not reflect the use of the struct. Memo is just the medium. Better to name the struct as ClaimMsg and use serde [rename](https://serde.rs/field-attrs.html#rename) for the memo field and add a comment that tells the data is extracted from memo field.

```rust
    #[serde(rename = "memo")]
    address: String,
```